core/sql/generator/GenRelUpdate.cpp

```diff
@@ -2652,7 +2656,8 @@ short HbaseInsert::codeGen(Generator *generator)
   queue_index downqueuelength = (queue_index)getDefault(GEN_DP2I_SIZE_DOWN);
   Int32 numBuffers = getDefault(GEN_DP2I_NUM_BUFFERS);
 
-  if (getInsertType() == Insert::VSBB_INSERT_USER)
+  if (getInsertType() == Insert::VSBB_INSERT_USER &&
+              generator->oltOptInfo()->multipleRowsReturned())
     downqueuelength = 400;
```

Comment:
If at some future point we could change this 400 to a cqd setting or a cardinality based setting or even a constant that will read better. No change is needed now.